language/tools/move-cli/src/base/movey_login.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use move_command_line_common::{env::MOVE_HOME, movey_constants::MOVEY_URL};
+use std::{fs, fs::File, io, path::PathBuf};
+use toml_edit::easy::{map::Map, Value};
+
+pub const MOVEY_CREDENTIAL_PATH: &str = "/movey_credential.toml";
+
+#[derive(Parser)]
+#[clap(name = "movey-login")]
+pub struct MoveyLogin;
+
+impl MoveyLogin {
+    pub fn execute(self) -> Result<()> {
+        println!(
+            "Please paste the API Token found on {}/settings/tokens below",
+            MOVEY_URL
+        );
+        let mut line = String::new();
+        loop {
+            match io::stdin().read_line(&mut line) {
+                Ok(_) => {
+                    line = line.trim().to_string();
+                    if !line.is_empty() {
+                        break;
+                    }
+                    println!("Invalid API Token. Try again!");
+                }
+                Err(err) => {
+                    bail!("Error reading file: {}", err);
+                }
+            }
+        }
+        Self::save_credential(line, MOVE_HOME.clone())?;
+        println!("Token for Movey saved.");
+        Ok(())
+    }
+
+    pub fn save_credential(token: String, move_home: String) -> Result<()> {
+        fs::create_dir_all(&move_home)?;
+        let credential_path = move_home + MOVEY_CREDENTIAL_PATH;
```

Comment:
Have you tested this on Windows? I have a feeling that this style of concatenating paths will not work there. It's used in other places too (e.g., in tests) - please check that all this works on Windows.

Reply:
We've tested both #226 and #227 and can confirm that they works.